crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -378,33 +380,32 @@ async fn sys_validation_workflow_inner(
             warrants.push(warrant_op);
         }
 
-        let warrant_op_hashes = warrants
-            .iter()
-            .map(|w| (w.as_hash().clone(), w.dht_basis()))
-            .collect::<Vec<_>>();
-
-        workspace
+        let authored_warrants = warrants.clone();
+        let warranted = workspace
             .authored_db
             .write_async(move |txn| {
-                for warrant_op in warrants {
+                let mut warranted = 0;
+                for warrant_op in authored_warrants {
                     insert_op_authored(txn, &warrant_op)?;
-                    summary.warranted += 1;
+                    warranted += 1;
+                }
+                StateMutationResult::Ok(warranted)
+            })
+            .await?;
+        // "self-publish" warrants, i.e. insert them into the DHT DB.
+        // This works around the problem that the commonly used function [`holochain_state::integrate::authored_ops_to_dht_db`]
+        // joins on the Action table to filter out private entries.
+        workspace
+            .dht_db
+            .write_async(move |txn| {
+                for warrant_op in warrants {
+                    insert_locally_validated_op(txn, warrant_op)?;
                 }
                 StateMutationResult::Ok(())
             })
             .await?;
 
-        if let Some(cache) = workspace.dht_query_cache.as_ref() {
-            // "self-publish" warrants, i.e. insert them into the DHT db as if they were published to us by another node
-            holochain_state::integrate::authored_ops_to_dht_db(
```

Comment:
This can't be used unfortunately, because it filters ops that have an action that have an entry that is private. Changing the query would have been a fine mess. The workaround above is not bad actually, as `insert_locally_validated_ops` is called by `authoted_ops_to_dht_db` after filtering.